slang-gfx.h

```diff
@@ -1769,9 +1770,23 @@ class ICommandQueue : public ISlangUnknown
 class ITransientResourceHeap : public ISlangUnknown
 {
 public:
+    struct Flags
+    {
+        enum Enum
+        {
+            None = 0,
+            AllowResizing = 0x1,
+        };
+    };
     struct Desc
     {
+        Flags::Enum flags;
         size_t constantBufferSize;
+        uint32_t samplerDescriptorCount;
+        uint32_t uavDescriptorCount;
+        uint32_t srvDescriptorCount;
+        uint32_t constantBufferDescriptorCount;
+        uint32_t accelerationStructureDescriptorCount;
```

Comment:
I'd like us to think about how to make these size hints be relatively API-independent *and* extensible to support new descriptor types as they come along.

Nothing that needs to happen today, of course.